aptos-move/framework/aptos-framework/sources/voting.spec.move

```diff
@@ -0,0 +1,195 @@
+spec aptos_framework::voting {
+    spec module {
+        pragma verify = true;
+        pragma aborts_if_is_strict;
+        invariant exists<timestamp::CurrentTimeMicroseconds>(@aptos_framework);
+    }
+
+    spec register<ProposalType: store>(account: &signer) {
+        let addr = signer::address_of(account);
+
+        // Will abort if there's already a `VotingForum<ProposalType>` under addr
+        aborts_if exists<VotingForum<ProposalType>>(addr);
+        // Creation of 4 new event handles changes the account's `guid_creation_num`
+        aborts_if !exists<account::Account>(addr);
+        let register_account = global<account::Account>(addr);
+        aborts_if register_account.guid_creation_num + 4 > MAX_U64;
+        // `type_info::type_of()` may abort if the type parameter is not a struct
+        aborts_if !type_info::spec_is_struct<ProposalType>();
+
+        ensures exists<VotingForum<ProposalType>>(addr);
+    }
+
+    /// The same as spec of `create_proposal_v2()`.
```

Comment:
Could we make the common spec of `create_proposal` and `create_proposal_v2` as a schema, and include it in both of the functions?